doc/ipythonconsole.rst

```diff
@@ -2,91 +2,145 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. _Connecting to a remote kernel: https://github.com/ipython/ipython/wiki/Cookbook:-Connecting-to-a-remote-kernel-via-ssh
+To connect to a kernel on a remote machine,
 
+#. Launch an IPython kernel on the remote host if one is not already running, with ``python -m spyder_kernels.console``.
 
-.. _console-features:
+#. Copy the kernel's connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) to the machine you're running Spyder on.
 
-==================
-Supported features
-==================
+   You can get :file:`{jupyter/runtime/dir/path}` by executing ``jupyter --runtime-dir`` in the same Python environment as the kernel.
+   Usually, the connection file you are looking for will be one of the newest in this directory, corresponding to the time you started the external kernel.
 
-Any :guilabel:`IPython Console` in Spyder, internally or externally created, supports additional features including:
+#. Click :guilabel:`Connect to an existing kernel` from the :guilabel:`Console` menu or the :guilabel:`IPython Console` pane's "Gear" menu.
 
-.. image:: images/console/console-completion.png
-   :alt: Spyder IPython Console, with a popup list of code completion guesses
+#. Browse for or enter the path to the connection file from the previous step.
 
-* Automatic code completion
-* Real-time function calltips
-* Debugging toolbar integration for launching the debugger and controlling execution flow
+   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to :file:`{jupyter/runtime/dir/path}/kernal-{id}.json` on your local machine, if you've copied the connection file there.
 
-Spyder-created consoles support even more advanced capabilities, such as:
+#. Check the :guilabel:`This is a remote kernel (via SSH)` box and type the full hostname you're connecting to (in the form :file:`{username}@{hostname}:{port-number}`).
+   Then, enter *either* :file:`{username}`'s password on the remote machine, or your user SSH keyfile (typically :file:`.perm`) (only one is needed to connect), and press :guilabel:`OK`.
 
-* The :doc:`variableexplorer`, with GUI-based editors for many built-in and third-party Python objects
-* Full GUI integration with the enhanced IPython debugger, ``ipdb``, including viewing and setting normal and conditional breakpoints interactively in any file, a :guilabel:`Breakpoints` pane, and following along with execution flow in the in the :doc:`editor` (see the :doc:`debugging` documentation for more details)
-* The :ref:`User Module Reloader <umr-section>`, which can automatically re-import modified packages and files
-* Inline display of Matplotlib graphics, if the ``Inline`` backend is selected under :menuselection:`Preferences --> IPython console --> Graphics --> Graphics backend`
+   The port number is the one on which the SSH daemon (``sshd``) is running, typically 22 unless you or your administrator has configured it otherwise.
```

Comment:
```suggestion
   The port number is the one on which the SSH daemon (``sshd``) on your server is running, typically 22 unless you or your administrator has configured it otherwise.
```

Small clarification.

Reply:
```suggestion
   :guilabel:`Port` is the port number on your remote machine that the SSH daemon (``sshd``) is listening on, typically 22 unless you or your administrator has configured it otherwise.
```

Some further clarification (its of course correct to call the remote machine a server in technical SSH terminology, but its best to avoid potentially confusing terms as the user might not consider it such and just use local and remote).